helm/config.go

```diff
@@ -0,0 +1,401 @@
+	return func(enc *Encoder) {
+		enc.emptyLines = emptyLines
+	}
+}
+
+// Indent sets the indentation amount per nesting level for the YAML encoding.
+// The default value is 2.
+func Indent(indent int) func(*Encoder) {
+	return func(enc *Encoder) {
+		if indent < 2 {
+			indent = 2
+		}
+		enc.indent = indent
+	}
+}
+
+// Wrap sets the maximum line length for comments. This number includes the
+// columns needed for indentation, so comments on more deeply nested nodes have
+// more tightly wrapped comments than outer level nodes. Wrapping applies only
+// to comments and not to block actions or scalar values. The default value is 80.
+func Wrap(wrap int) func(*Encoder) {
+	return func(enc *Encoder) {
+		enc.wrap = wrap
+	}
+}
+
+// Set applies Encoder modifier functions to set option values.
+func (enc *Encoder) Set(modifiers ...func(*Encoder)) {
+	for _, modifier := range modifiers {
+		modifier(enc)
+	}
+}
+
+// NewEncoder returns an Encoder object holding the output stream and encoding options.
+func NewEncoder(writer io.Writer, modifiers ...func(*Encoder)) *Encoder {
+	enc := &Encoder{
+		writer:     writer,
+		err:        nil,
+		emptyLines: false,
+		indent:     2,
+		wrap:       80,
+	}
+	enc.Set(modifiers...)
+	return enc
+}
+
+// Encode writes the config object to the stream.
+func (enc *Encoder) Encode(object *Object) error {
+	enc.pendingNewline = false
+	fmt.Fprintln(enc, "---")
+	prefix := ""
+	enc.writeNode(object, &prefix, "")
+	return enc.err
+}
+
+// Write implements the io.Writer interface. It just forwards to the embedded
+// writer until an error occurs. This allows for error checking just once at the
+// end of Encode().
+func (enc *Encoder) Write(buffer []byte) (int, error) {
+	if enc.err == nil {
+		var written int
+		written, enc.err = enc.writer.Write(buffer)
+		return written, enc.err
+	}
+	return 0, enc.err
+}
+
+// useOnce returns the current value of the prefix parameter but replaces it
+// with a string of spaces of the same length as the original prefix for
+// subsequent use. This is done because for nested list elements the "- "
+// prefixes are being stacked, and should only be used in front of the first
+// element of the innermost list:
+//
+//   [ [ [ 1, 2 ] ] ]   --->   " - - - 1\n    - 2\n"
+//
+func useOnce(prefix *string) string {
+	result := *prefix
+	*prefix = strings.Repeat(" ", len(*prefix))
+	return result
+}
+
+// writeComment writes out the comment lines for a node. Newline characters in
+// the comment mark the beginning of a new paragraph. Each paragraph is
+// word-wrapped to fit within enc.wrap columns (except that each line will
+// include at least a single word, even if it exceeds the wrapping column).
+func (enc *Encoder) writeComment(prefix *string, comment string) {
+	for _, line := range strings.Split(comment, "\n") {
+		fmt.Fprintf(enc, "%s#", useOnce(prefix))
+		if len(line) > 0 {
+			written := 0
+			for _, word := range strings.Fields(line) {
```

Comment:
I guess this actually breaks multiple spaces in sequence (e.g. if somebody was trying to draw a diagram). Do we care?

Reply:
You can't ask for line wrapping *and* also want to preserve inline ascii art.

Well you can, just make sure your lines are short enough, so that they don't get wrapped. 😄

Reply:
But we wrap _all_ lines here, don't we? Even the really short ones?